src/main/java/com/synopsys/integration/alert/channel/azure/boards/actions/AzureBoardsGlobalApiAction.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.channel.azure.boards.descriptor.AzureBoardsDescriptor;
+import com.synopsys.integration.alert.common.action.ApiAction;
+import com.synopsys.integration.alert.common.descriptor.DescriptorKey;
+import com.synopsys.integration.alert.common.descriptor.DescriptorMap;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.persistence.accessor.ConfigurationAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
+import com.synopsys.integration.alert.common.persistence.util.ConfigurationFieldModelConverter;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.FieldValueModel;
+
+@Component
+public class AzureBoardsGlobalApiAction extends ApiAction {
+    private ConfigurationAccessor configurationAccessor;
+    private DescriptorMap descriptorMap;
+    private ConfigurationFieldModelConverter fieldModelConverter;
+
+    @Autowired
+    public AzureBoardsGlobalApiAction(ConfigurationAccessor configurationAccessor, DescriptorMap descriptorMap, ConfigurationFieldModelConverter configurationFieldModelConverter) {
+        this.configurationAccessor = configurationAccessor;
+        this.descriptorMap = descriptorMap;
+        this.fieldModelConverter = configurationFieldModelConverter;
+    }
+
+    @Override
+    public FieldModel beforeSaveAction(FieldModel fieldModel) throws AlertException {
+        FieldModel updatedFieldModel = super.beforeSaveAction(fieldModel);
+        Optional<DescriptorKey> descriptorKey = descriptorMap.getDescriptorKey(fieldModel.getDescriptorName());
+
+        if (descriptorKey.isPresent()) {
+            ConfigContextEnum context = ConfigContextEnum.valueOf(fieldModel.getContext());
+            List<ConfigurationModel> existingConfig = configurationAccessor.getConfigurationsByDescriptorKeyAndContext(descriptorKey.get(), context);
+            Optional<ConfigurationModel> configurationModel = existingConfig.stream()
+                                                                  .findFirst();
+            updatedFieldModel = updateTokenFields(updatedFieldModel, configurationModel);
+        }
+        return updatedFieldModel;
+    }
+
+    @Override
+    public FieldModel beforeUpdateAction(FieldModel fieldModel) throws AlertException {
+        FieldModel updatedFieldModel = super.beforeUpdateAction(fieldModel);
+        Optional<ConfigurationModel> existingConfig = configurationAccessor.getConfigurationById(Long.valueOf(fieldModel.getId()));
+        return updateTokenFields(updatedFieldModel, existingConfig);
+    }
+
+    private FieldModel updateTokenFields(FieldModel fieldModel, Optional<ConfigurationModel> existingConfig) {
```

Comment:
I think this method should take in only a ConfigurationModel and return a Map<String, FieldValueModel> object. Then wherever we generate the Optional, we instead call map and pass this method as a method reference. Then just return the new object like we do on line 93. Does that make sense?